planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/include/autoware/behavior_path_goal_planner_module/pull_over_planner_base.hpp

```diff
@@ -52,41 +52,28 @@ struct PullOverPath
   size_t goal_id{};
   size_t id{};
   bool decided_velocity{false};
-  double max_curvature{0.0};  // max curvature of the parking path
 
-  PathWithLaneId getFullPath() const
+  /**
+   * @brief Set paths and start/end poses
+   * By setting partial_paths, full_path, parking_path and curvature are also set at the same time
+   * @param input_partial_paths partial paths
+   * @param input_start_pose start pose
+   * @param input_end_pose end pose
+   */
+  void setPaths(
+    const std::vector<PathWithLaneId> input_partial_paths, const Pose & input_start_pose,
+    const Pose & input_end_pose)
   {
-    PathWithLaneId path{};
-    for (size_t i = 0; i < partial_paths.size(); ++i) {
-      if (i == 0) {
-        path.points.insert(
-          path.points.end(), partial_paths.at(i).points.begin(), partial_paths.at(i).points.end());
-      } else {
-        // skip overlapping point
-        path.points.insert(
-          path.points.end(), next(partial_paths.at(i).points.begin()),
-          partial_paths.at(i).points.end());
-      }
-    }
-    path.points = autoware::motion_utils::removeOverlapPoints(path.points);
+    partial_paths = input_partial_paths;
+    start_pose = input_start_pose;
+    end_pose = input_end_pose;
 
-    return path;
+    updatePathData();
   }
 
-  // path from the pull over start pose to the end pose
-  PathWithLaneId getParkingPath() const
-  {
-    const PathWithLaneId full_path = getFullPath();
-    const size_t start_idx =
-      autoware::motion_utils::findNearestIndex(full_path.points, start_pose.position);
+  PathWithLaneId getFullPath() { return full_path; }
```

Comment:
We should return const& reference

Reply:
fixed in dbd9ff70604526c6c8b8ef5b142a31c112c0a724

Reply:
revert in [use copy getter](https://github.com/autowarefoundation/autoware.universe/pull/8195/commits/8ce336e0a1d1dddfb2ce7a10e82b97cab7685401)

becase the returned value is used in multi threads